src/bb_excel/core.clj

```diff
@@ -130,127 +100,178 @@
-          row*)))
+      (= cell-type "s")                 (nth shared-strings (parse-long raw-cell-value))
+      (= cell-type "str")               raw-cell-value
+      (= cell-type "inlineStr")         raw-cell-value
+      (= cell-type "b")                 (if (= "1" raw-cell-value) true false)
+      (= cell-type "e")                 (get error-codes raw-cell-value)
+      (= cell-type "n")                 (parse-long raw-cell-value)
+      (style-check cell-attrs styles pcts)    (num2pct raw-cell-value)
+      (style-check cell-attrs styles dates)   (num2date raw-cell-value)
+      (style-check cell-attrs styles times)   (num2time raw-cell-value)
+      :else raw-cell-value)))
 
 (defn- get-cell-text
   "Extract text from cell"
   [cell]
   (->> (xml-seq cell)
-       (filter #(contains? TEXT_T_TAGS (:tag %)))
+       (filter (by-tag :t))
        (mapcat :content)
        (str/join)))
 
 (defn get-shared-strings
   "Get dictionary of all unique strings in the Excel spreadsheet"
   [^ZipFile zipfile]
-  (if-let [wb (.getEntry zipfile "xl/sharedStrings.xml")]
-    (let [ins (.getInputStream zipfile wb)
-          x (parse-str (slurp ins))]
-      (into [] (comp (filter #(contains? TEXT_PART_TAGS (:tag %)))
-                     (map get-cell-text))
-            (xml-seq x)))
+  (if-let [shared-strings-entry (.getEntry zipfile "xl/sharedStrings.xml")]
+    (with-open [shared-strings (.getInputStream zipfile shared-strings-entry)]
+      (let [sst-node (xml/parse shared-strings {:namespace-aware false})]
+        (mapv get-cell-text (:content sst-node))))
     []))
 
 (defn get-styles
-  "Get styles"
   [^ZipFile zipfile]
-  (if-let [wb (.getEntry zipfile "xl/styles.xml")]
-    (let [ins (.getInputStream zipfile wb)
-          x   (parse-str (slurp ins))]
-      (->> (xml-seq x)
-           (filter #((:cellxfs tags) (:tag %)))
-           first
-           :content
-           (filter #((:xf tags) (:tag %)))
-           (mapv (comp :numFmtId :attrs))))
+  (if-let [styles-entry (.getEntry zipfile "xl/styles.xml")]
+    (with-open [styles (.getInputStream zipfile styles-entry)]
+      (let [style-sheet-node (xml/parse styles {:namespace-aware false})
+            cell-xfs-node (->> (:content style-sheet-node)
+                               (find-first (by-tag :cellXfs)))
+            xf-nodes (->> (:content cell-xfs-node)
+                          (filter (by-tag :xf)))]
+        (mapv #(-> % :attrs :numFmtId) xf-nodes)))
     []))
 
+(def ^:const BASE_ROW_INDEX 0)
+(def ^:const START_COLUMN_INDEX 1)
+
+(defn valid-cell-index?
+  [cell-index]
+  (if cell-index
+    (boolean (re-find #"^[A-Z]{1,3}\d+$" cell-index))
+    false))
+
+(def ^:const A_CHAR_INDEX (int \A))
+
+(defn number->column-letter
+  [n]
+  (loop [num n
+         acc ""]
+    (if (> num 0)
+      (let [residue (mod (dec num) 26)
+            new-num (quot (dec num) 26)]
+        (recur new-num (str (char (+ residue A_CHAR_INDEX)) acc)))
+      acc)))
```

Comment:
With this functionality can add new option to retrieve rows with number cell index If this may be of interest to anyone